packages/vite-plugin-svelte/src/plugins/configure.js

```diff
@@ -87,6 +94,12 @@ export function configure(api, inlineOptions) {
 			ensureConfigEnvironmentConditions(name, config, opts);
 			// @ts-expect-error the function above should make `resolve.conditions` non-nullable
 			config.resolve.conditions.push('svelte');
+			if (config.consumer === 'server' && optimizeDeps !== undefined) {
+				// optimizeDeps is not inherited by server environments so return it here
+				return {
+					optimizeDeps
```

Comment:
not entirely happy with this for 2 reasons

1) this configuration part should probably be moved out of this plugin and into the setup-optimizer plugin, but that requires a bit of refactoring and maybe running vitefu twice.

2) it replicates the same optimizeDeps for server and client env, but i am not sure yet if the same rules apply

Reply:
> it replicates the same optimizeDeps for server and client env, but i am not sure yet if the same rules apply

I was thinking the same thing. Would it be enough to just pass `extensions` and `plugins` so that esbuild knows how to handle Svelte modules? Then the user can decide what to include/exclude for optimizing themselves.